Resources/config/jackalope_doctrine_dbal.xml

```diff
@@ -13,15 +13,15 @@
 
         <service id="doctrine_phpcr.jackalope_doctrine_dbal.schema"
                  class="%doctrine_phpcr.jackalope_doctrine_dbal.repository_schema.class%"
-                 public="false"
+                 public="true"
```

Comment:
Wouldn't marking that service as lazy (lazy="true") have the same effect?

Reply:
We would need to include a dependency on `ocramius/proxy-manager` to do this. https://symfony.com/doc/3.4/service_container/lazy_services.html

Reply:
@eddmann I tested it the listener itselfs are lazy if the proxy-manager is installed you see also a lazy 1 in this service tag:

> bin/console debug:container doctrine_phpcr.jackalope_doctrine_dbal.schema_listener
>   Tags             doctrine.event_listener (connection: , event: postGenerateSchema, lazy: 1)

So I would avoid this change and if somebody need to have them lazy in that project need to install the proxy-manager by itself would not make it as a requirement of using phpcr bundle.

Reply:
Okay but this now differs to what has been merged at https://github.com/doctrine/DoctrinePHPCRBundle/pull/338 now

Reply:
Oh didn't know that there was something merged. I would revert because as mention it is a BC Break also when a constructor is changed.

Reply:
i considered this class internal - is there a use case to extend it?

from that doc:

> If you don't install the ProxyManager bridge and the ocramius/proxy-manager, the container will just skip over the lazy flag and instantiate the service as it would normally do.

how about we mark the lazy and document that you need proxy-manager bridge if you want to avoid the database hit?